### PR TITLE
DAT-2973 hide caption for videos in infoboxes

### PR DESCRIPTION
**Pull request by dianafa**
https://wikia-inc.atlassian.net/browse/DAT-2973

UWAGA!!!
Has to be pushed together with the app part: https://github.com/Wikia/app/pull/7768

### CODE REVIEW

File: front/scripts/main/components/VideoMediaComponent.ts [hunk 1]
```diff
@@ -4,5 +4,10 @@
...
 App.VideoMediaComponent = App.ImageMediaComponent.extend({
 	classNames: ['article-video'],
-	layoutName: 'components/video-media'
+	layoutName: 'components/video-media',
+
+	isInfoboxVideo: Em.computed('media', function (): boolean {
+		var media = this.get('media');
+		return media.context === 'infobox-video';
+	})
 });
```
Comment (dianafa) [inline front/scripts/main/components/VideoMediaComponent.ts]: it's nicer to create this prop and check if is true than setting the `media.title` and `caption` to null

File: front/scripts/main/components/VideoMediaComponent.ts [hunk 2]
```diff
@@ -4,5 +4,10 @@
 
 App.VideoMediaComponent = App.ImageMediaComponent.extend({
 	classNames: ['article-video'],
-	layoutName: 'components/video-media'
+	layoutName: 'components/video-media',
+
+	isInfoboxVideo: Em.computed('media', function (): boolean {
+		var media = this.get('media');
+		return media.context === 'infobox-video';
```
Comment (rogatty) [inline front/scripts/main/components/VideoMediaComponent.ts]: `isInfoboxVideo: Em.computed.equal('media.context', 'infobox-video')`

Comment (dianafa) [inline front/scripts/main/components/VideoMediaComponent.ts]: dziękuję :)

File: front/scripts/main/components/MediaComponent.ts [hunk 3]
```diff
@@ -65,7 +65,7 @@ App.MediaComponent = Em.Component.extend(App.VisibleMixin, {
 		get(): string {
 			var media = this.get('media');
 
-			if (media && media.context !== 'infobox-image' && typeof media.caption === 'string') {
+			if (media && media.context !== 'infobox-image' && media.context !== 'infobox-video' && typeof media.caption === 'string') {
```
Comment (SebastianMarzjan) [inline front/scripts/main/components/MediaComponent.ts]: Did you run tslint? This line seems to violate the [guidelines](https://github.com/Wikia/guidelines/blob/master/TypeScript/tslint.json) in regard to line length. Maybe it'd make sense to extract infobox-specific checks at this point?

File: front/templates/main/components/video-media.hbs [hunk 4]
```diff
@@ -1,5 +1,7 @@
 <img src='{{imageSrc}}' style='{{style}}'>
-<figcaption>
-	<div>{{media.title}}</div>
-	{{{caption}}}
-</figcaption>
+{{#unless isInfoboxMedia}}
```
Comment (RafLeszczynski) [inline front/templates/main/components/video-media.hbs]: consider passing more general variable name to the template (for example `noCaption`)

**Comment by RafLeszczynski** [comment]
LGTM

**Pull request merged**